TestCases/TestCase.py

```diff
@@ -325,7 +334,90 @@ def run_filediff(self):
+                        ignore_counter  = 0
+
+                        # Assert that both files have the same number of lines
+                        if len(fromlines) != len(tolines):
+                            diff   = ["ERROR: Number of lines in " + fromfile + " and " + tofile + " differ."]
+                            passed = False
+                        
+                        # Loop through all lines
+                        for i_line in range(0, len(fromlines)):
+
+                            if passed == False: break
+
+                            # Extract next line and split it
+                            from_line = fromlines[i_line].split()
+                            to_line   = tolines[i_line].split()
+
+                            # Assert that both lines have the same number of words
+                            if len(from_line) != len(to_line):
+                                diff   = ["ERROR: Number of words in line " + str(i_line+1) + " differ."]
+                                passed = False
+                                break
+                            
+                            # Loop through all words of one line
+                            for i_word in range(0, len(from_line)):
+
+                                # Extract next word and strip whitespace and commas
+                                from_word = from_line[i_word].strip().strip(',')
+                                to_word   = to_line[i_word].strip().strip(',')
+
+                                # Assert that words are either both numeric or both non-numeric
+                                from_isfloat = is_float(from_word)
+                                to_isfloat   = is_float(to_word)
+                                if from_isfloat != to_isfloat:
+                                    diff      = ["ERROR: File entries '" + from_word + "' and '" + to_word + "' in line " + str(i_line+1) + ", word " + str(i_word+1) + " differ."]
+                                    passed    = False
+                                    delta     = 0.0
+                                    max_delta = "Not applicable"
+                                    break
+
+                                # Make actual comparison
+                                # Compare floats
+                                if from_isfloat:
+                                    try:
+                                        # Only do a relative comparison when the threshold is met.
+                                        # This is to prevent large relative differences for very small numbers.
+                                        if (abs(float(from_word)) > self.comp_threshold):
+                                            delta = abs( (float(from_word) - float(to_word)) / float(from_word) ) * 100
+                                            compare_counter += 1
+                                        else:
+                                            delta = 0.0
```

Comment:
Should we compare absolute values in this case? If all values happen to be small this could pass everything.

Reply:
Thanks for your review and comments Pedro. The main reason to compare relative values is that we want to run this comparison on restart files that hold quantities that vary on very different ranges. E.g., enthalpy might range on O(1e6) while progress variable typically ranges on O(1). I am not sure if specifying the same tolerance for these quantities for an absolute comparison would make much sense as a fall back solution. But we can definitely add it as an alternative for files that are better suited for absolute comparisons. What do you think?

Regarding your comment on CodeQL warnings: where do I find those? Do you mean the issues found by CodeFactor?

Reply:
I meant when the values are below the tolerance, instead of assuming 0, doing an absolute comparison. But I understand the motivation.
Yes CodeFactor, if it is only complaining about white spaces it's fine.

Reply:
The value of "delta" will later be used to check if the compared value is within the specified relative tolerance. If we assign the absolute difference here it gets inconsistent. My line of thinking was that it's most straightforward to ignore values below the threshold and make the developer, who hooks up his test case, responsible for specifying reasonable values for tolerance and threshold. As a guide, we report the number of ignored and compared file entries, as well as the maximum difference at the end of the test. Do you think that's ok to ask of the other developers? 

I removed some white space and a semicolon which CodeFactor didn't like.

Reply:
Sounds good